### DISCUSSION TITLE
New feature request: Stepped Controlnet

### DISCUSSION THREAD

**Discussion by pflky** [Mikubill/sd-webui-controlnet#657] [Ideas]
I'm not looking for this feature personally, but it seems like a feature that would help solve a problem a lot of people have, being that they like normal non-controlnet generations at around X number of steps, but then the generation goes past that and gives less desirable results. So the solution seems to be the ability to automatically run controlnet after X number of steps, and then to let the generation keep working until finish, with those details locked in.  This could be used to also apply an img2img at the same time. It would require a few new checkboxes and sliders for setting controlnet to automatically activate after a set number of steps, and whether to employ an img2img or not, along with things like denoising strength for when it uses img2img in this way. 

Now of course a person could just do all this manually, while just limiting the number of steps, but I think his has the potential to add something to generation quality if controlnet could be applied mid-generation and then the generation finishes the rest of the steps with these controls in place.

**Comment by ljleb** [comment]
I think this has already been implemented with guidance start and guidance end? If I understand correctly your suggestion.

**Reply by pflky** [reply]
Perhaps? Not sure. But controlnet generally requires a starting image. What I mean is for even just txt2img generations, where it would automatically run controlnet after X number of steps, without having to load any images manually. Then either finish the txt2img generation with the remaining number of steps, or re-run from the start with the full number of steps.
